### PR TITLE
fix(TPG >= 4.53)!: set bgp keepalive_interval to default 20

### PR DESCRIPTION
**Pull request by dlouvier**
issue: https://github.com/terraform-google-modules/terraform-google-vault/issues/171
---
`keepalive_interval` is not set and this value needs to be between 20 and 60.

> If set, this value must be between 20 and 60. The default is 20.

[Provider documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_router#keepalive_interval)

As this is not set, every time we run `terraform plan`, the plan outputs this change and it is quite noisy.

image

### CODE REVIEW

**Comment by dlouvier** [comment]
Hello @cft-admins could you run /gcbrun ?

Thanks

File: network.tf [hunk 1]
```diff
@@ -53,6 +53,7 @@ resource "google_compute_router" "vault-router" {
 
   bgp {
     asn = 64514
+    bgp = 20
```
Comment (apeabody) [inline network.tf:56]: Should this be `keepalive_interval`?

```suggestion
    keepalive_interval = 20
```

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @dlouvier! - Can you PTAL?

**Comment by apeabody** [comment]
Looks like `keepalive_interval` was added in TPG 4.53.0, would need to also bump https://github.com/terraform-google-modules/terraform-google-vault/blob/master/versions.tf#L23

**Comment by apeabody** [comment]
Thanks @dlouvier! - Some relevant output from the LINT tests:

```
terraform_validate . 
╷
│ Error: Unsupported argument
│ 
│   on network.tf line 56, in resource "google_compute_router" "vault-router":
│   56:     bgp = 20
│ 
│ An argument named "bgp" is not expected here.
╵
terraform_validate ./examples/shared_vpc_internal 
Success! The configuration is valid.

terraform_validate ./examples/vault-on-gce 
╷
│ Error: Unsupported argument
│ 
│   on ../../network.tf line 56, in resource "google_compute_router" "vault-router":
│   56:     bgp = 20
│ 
│ An argument named "bgp" is not expected here.
╵
terraform_validate ./modules/cluster 
Success! The configuration is valid.

terraform_validate ./test/fixtures/simple_external 
╷
│ Error: Unsupported argument
│ 
│   on ../../../network.tf line 56, in resource "google_compute_router" "vault-router":
│   56:     bgp = 20
│ 
│ An argument named "bgp" is not expected here.
╵
terraform_validate ./test/setup 
```

**Comment by dlouvier** [comment]
Sure, thanks for your fix @apeabody

**Comment by apeabody** [comment]
Hi @dlouvier - Looks like just needs a `terraform fmt`:

```
Running terraform fmt
network.tf
--- old/network.tf
+++ new/network.tf
@@ -52,7 +52,7 @@
   network = local.network
 
   bgp {
-    asn = 64514
+    asn                = 64514
     keepalive_interval = 20
   }
 
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
```

File: network.tf [hunk 2]
```diff
@@ -53,6 +53,7 @@ resource "google_compute_router" "vault-router" {
 
   bgp {
     asn = 64514
```
Comment (dlouvier) [inline network.tf:55]:
```suggestion
    asn                = 64514
```

**Comment by dlouvier** [comment]
@apeabody hey sorry, I did abandoned this. Fixed the format (I was editing from the GitHub UI and yep... it seemed straight forward but it wasn't).

Can you run tests again ? Should be good now.

**Comment by dlouvier** [comment]
🚀

**Review by apeabody — COMMENTED** [review]
Thanks for the contribution @dlouvier!

@bharathkkb - This might cause "churn" during first application, but as it's already a breaking change for TPG I think that should be OK?

**Comment by dlouvier** [comment]
Any update folks,

@apeabody @bharathkkb 
Can you run again `/gcbrun` ? 

I've rebased the PR as there was a commit in the main branch.

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @dlouvier!

**Comment by dlouvier** [comment]
you welcome, see ya around!

**Pull request merged**